system/jlib/jsecrets.cpp

```diff
@@ -387,13 +432,24 @@ class SecretCacheEntry : public CInterface
         return contentHash;
     }
 
+    void getSecretOptions(std::string & category, std::string & name, std::string & optVaultId, std::string & optVersion)
+    {
+        expandSecretKey(category, name, optVaultId, optVersion, secretKey.c_str());
+    }
+
     // We should never replace known contents for unknown contents
     // so once this returns true it should always return true
     bool hasContents() const
     {
         return contents != nullptr;
     }
 
+    //Has the secret value been used since it was last checked for an update?
```

Comment:
I'm struggling to understand why it is relevant whether a secret value has been used recently, when updating them. Unless you are removing secrets from the cache if not used recently (which I don't think you are).

Reply:
The aim is to not refresh secrets in the background that have not been used recently - to avoid unnecessary load on the vault.